### PR TITLE
process-util: add helper get_process_threads()

### PR DESCRIPTION
**Pull request by poettering**
Let's add a proper helper for querying the number of threads in a process.

### CODE REVIEW

**Review by DaanDeMeyer — APPROVED** [review]

File: src/basic/process-util.c [hunk 1]
```diff
@@ -1549,6 +1549,31 @@ _noreturn_ void freeze(void) {
...
+        r = get_proc_field(p, "Threads", WHITESPACE, &t);
+        if (r == -ENOENT)
+                return proc_mounted() == 0 ? -ENOSYS : -ESRCH;
+        if (r < 0)
+                return r;
+
+        r = safe_atoi(t, &n);
+        if (r < 0)
+                return r;
+        if (n < 0)
```
Comment (yuwata) [inline src/basic/process-util.c:1571]: Is it possible `n == 0`?? How about `if (n <= 0)` ?

Comment (poettering) [reply src/basic/process-util.c:1571]: yes, n== 0 is possible for zombie processes. i.e. processes which died and thus have no tasks anymore, but which haven't been reaped by their parent process yet, and thus still exists as a PID and in /proc/.

Comment (yuwata) [reply src/basic/process-util.c:1571]: Oh, I did not know that. Interesting. Thank you.

**Comment by poettering** [comment]
Set green label again. I finally made the testcase work. Apparently the data from /proc/ is updated asynchronously, hence we must add manual sync points where we can, and be a bit less strict.

I only changed the test since @DaanDeMeyer greenlit it originally, hence taking liberty to set it back.

**Review by yuwata — APPROVED** [review]
LGTM.

**Pull request merged**